Libraries/PeriphDrivers/Source/GPIO/gpio_me30.c

```diff
@@ -100,50 +100,36 @@ int MXC_GPIO_Config(const mxc_gpio_cfg_t *cfg)
     }
 
     // Configure the pad
-    // TODO(ME30): "ps" (weak vs strong pull-up/down select) register field missing
     switch (cfg->pad) {
-    case MXC_GPIO_PAD_NONE:
-        gpio->padctrl0 &= ~cfg->mask;
-        gpio->padctrl1 &= ~cfg->mask;
-        break;
-
-    case MXC_GPIO_PAD_WEAK_PULL_UP:
-        gpio->padctrl0 |= cfg->mask;
-        gpio->padctrl1 &= ~cfg->mask;
-        // gpio->ps &= ~cfg->mask;
-        break;
-
-    case MXC_GPIO_PAD_PULL_UP:
-        gpio->padctrl0 |= cfg->mask;
-        gpio->padctrl1 &= ~cfg->mask;
-        // gpio->ps |= cfg->mask;
-        break;
-
-    case MXC_GPIO_PAD_WEAK_PULL_DOWN:
-        gpio->padctrl0 &= ~cfg->mask;
-        gpio->padctrl1 |= cfg->mask;
-        // gpio->ps &= ~cfg->mask;
-        break;
-
-    case MXC_GPIO_PAD_PULL_DOWN:
-        gpio->padctrl0 &= ~cfg->mask;
-        gpio->padctrl1 |= cfg->mask;
-        // gpio->ps |= cfg->mask;
-        break;
-
-    default:
-        return E_BAD_PARAM;
+        // Configure the pad
+        switch (cfg->pad) {
+        case MXC_GPIO_PAD_NONE:
+            gpio->padctrl0 &= ~cfg->mask;
+            break;
+
+        case MXC_GPIO_PAD_PULL_UP:
+            gpio->padctrl0 |= cfg->mask;
+            gpio->pssel |= cfg->mask;
+            break;
+
+        case MXC_GPIO_PAD_PULL_DOWN:
+            gpio->padctrl0 |= cfg->mask;
+            gpio->pssel &= ~cfg->mask;
+            break;
+
+        default:
+            return E_BAD_PARAM;
+        }
+
+        // Configure the drive strength
+        if (cfg->func == MXC_GPIO_FUNC_IN) {
+            return E_NO_ERROR;
+        } else {
+            return MXC_GPIO_SetDriveStrength(gpio, cfg->drvstr, cfg->mask);
+        }
```

Comment:
This if else block should be out of switch case
![image](https://github.com/analogdevicesinc/msdk/assets/46590392/b36c6c31-7e16-44e3-a721-ea55dfae5d22)